src/modules/components/ensUserLookupWrapper/ensUserLookupWrapper.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState, type ReactNode } from 'react';
+import { isAddress, type Hash } from 'viem';
+import { useEnsAddress, useEnsAvatar, useEnsName } from 'wagmi';
+// @ts-expect-error - untyped module
+import makeBlockiesUrl from 'blockies-react-svg/dist/es/makeBlockiesUrl.mjs';
+
+export interface IENSUser {
+    name: string;
+    address: string;
+    avatar: string;
+}
+
+export interface IENSLookupProps {
+    ensNameOrAddress: string;
+    children: (data: { user: IENSUser | null; error: Error | null; isLoading: boolean }) => ReactNode;
+}
+
+const ENSLookupComponent: React.FC<IENSLookupProps> = ({ ensNameOrAddress, children }) => {
```

Comment:
I wouldn't introduce such utility for now, the wagmi hooks are already pretty easy to reuse and for now we don't plan to have ens/address resolving logic on components rather than this one and the address-input component.
I would instead update the memberAvatar component to accept these props:
```
- address?: string
- ensName?: string
- avatarSrc?: string
```
And implement a logic such:
- If `avatarSrc` is set, just display it without resolving the ens name / avatar;
- If `ensName` is set, resolve ensAvatar and display it if valid or fallback to blockies;
- If `address` is set and is a valid address, resolve ensName and ensAvatar and display it if valid or fallback to blickies;
- If `address` is not set or is not a valid address, display the default avatar fallback (the grey one). This use-case is new and we just discussed it with Selim.

The logic above can be easily implemented by using the hooks from wagmi and the `enabled` property like:
```
[memberAvatar.tsx]

import { isAddress} from 'viem';
[...]
const { address, ensName, avatarSrc } = props;
const isValidAddress = address != null && isAddress(address);

const { data: ensData } = useEnsName({ address: address, query: { enabled: avatarSrc != null && ensName != null && isValidAddress } });
const processedEnsName = ensName ?? ensData;

const { data: ensAvatar } = useEnsAvatar({ name: processedEnsName, query: { enabled: avatarSrc != null && processedEnsName != null } });
const processedAvatar = avatarSrc ?? ensAvatar;

const fallback = isValidAddress ? [blockies-url-from-address] : undefined;

return <Avatar src={processedAvatar} fallback={fallback} />
```

Let me know if this approach is clear or if we need to sync on it :)